ext/CEGUI/IrrlichtBaWTexture.cpp

```diff
@@ -148,17 +237,52 @@ void IrrlichtBaWTexture::loadFromMemory(const void* buffer, const CEGUI::Sizef&
     viewParams.subresourceRange.levelCount = 1;
     viewParams.viewType = decltype(viewParams.viewType)::ET_2D;
 
-    gpuTexture = driver->createGPUImageView(std::move(viewParams));
+    frameBuffer = createFrameBuffer(driver->createGPUImageView(std::move(viewParams)));
 }
 
 void IrrlichtBaWTexture::blitFromMemory(const void* sourceData, const CEGUI::Rectf& area)
 {
+    bool status = frameBuffer;
 
+    if (status)
+    {
+        auto gpuTexture = frameBuffer->getAttachment(textureColorAttachment);
+        auto& viewParams = gpuTexture->getCreationParameters();
+        auto& extent = viewParams.image->getCreationParameters().extent;
+        irr::core::recti coverArea(area.left(), area.top(), area.getWidth(), area.getHeight());
+
+        auto temporaryInFrameBuffer = createFrameBuffer(sourceData, viewParams.format, extent.width, extent.height);
+        driver->blitRenderTargets(temporaryInFrameBuffer, frameBuffer, false, false, coverArea, coverArea);
+        temporaryInFrameBuffer->drop();
+    }
+    else
+        assert(status);
 }
 
 void IrrlichtBaWTexture::blitToMemory(void* targetData)
 {
+    bool status = frameBuffer;
+
+    if (status)
+    {
+        auto gpuTexture = frameBuffer->getAttachment(textureColorAttachment);
+        auto& viewParams = gpuTexture->getCreationParameters();
+        auto& extent = viewParams.image->getCreationParameters().extent;
+
+        auto temporaryOutFrameBuffer = createFrameBuffer(targetData, viewParams.format, extent.width, extent.height);
+        driver->blitRenderTargets(frameBuffer, temporaryOutFrameBuffer, false, false);
+ 
+        auto bufferByteSize = viewParams.image->getImageDataSizeInBytes();
+        auto destinationBoundMemory = const_cast<video::IDriverMemoryAllocation*>(viewParams.image->getBoundMemory()); // not sure
+        destinationBoundMemory->mapMemoryRange(video::IDriverMemoryAllocation::EMCAF_READ, { 0u, bufferByteSize });
```

Comment:
that's not how it works and it won't work this way.

Reply:
you need `copyImageToBuffer`

Reply:
do we implement `copy` for gpu objects?

Reply:
adjusted in following commit